src/main/java/com/artipie/rpm/meta/XmlPrimary.java

```diff
@@ -438,15 +431,31 @@ public Format headerRange(final int start, final int end) throws XMLStreamExcept
 
         /**
          * Add list of providers.
-         * @param providers Provider names
+         * @param name Provides names
+         * @param versions Versions
          * @return Self
          * @throws XMLStreamException On XML error
+         * @todo #220:30min We need to experimentally check that names and versions do not
+         *  mix up in `provides`: download oxygen-gtk2-1.3.4-3.el7.x86_64.rpm from test bundle100
+         *  (see TestBundle.class), add it to resources folder and create xml primary from it.
+         *  This rpm has not full `provides` information:
+         *  <rpm:entry name="liboxygen-gtk.so()(64bit)"/>
+         *  <rpm:entry name="oxygen-gtk2" flags="EQ" epoch="0" ver="1.3.4" rel="3.el7"/>
+         *  <rpm:entry name="oxygen-gtk2(x86-64)" flags="EQ" epoch="0" ver="1.3.4" rel="3.el7"/>
+         *  Make sure we write it the same way. Also we need
```

Comment:
@olenagerasimova seems like something is missing at the end of the puzzle, please take a look